app/scripts/controllers/edit/autoscaler.js

```diff
@@ -78,6 +78,21 @@ angular.module('openshiftConsole')
     var horizontalPodAutoscalerVersion = APIService.getPreferredVersion('horizontalpodautoscalers');
     var limitRangesVersion = APIService.getPreferredVersion('limitranges');
 
+    var getHPAWarnings = function(resource, limitRanges, project, context) {
+      var scaleTarget = resource.spec.scaleTargetRef;
+      var scaleTargetVersion = APIService.objectToResourceGroupVersion(scaleTarget);
+      // if the DC still has extensions in the verison, it will fail, so we need to
+      // get the preferred version.
```

Comment:
Wish we had es6 fat arrows && auto `return` 😄

Reply:
This gets the `edit` `autoscaler` page able to use `warnings` from the `hpaService` instead of the standalone `showCPURequestWarning` flags.